presto-cli/src/main/java/io/prestosql/cli/OutputHandler.java

```diff
@@ -13,85 +13,79 @@
  */
 package io.prestosql.cli;
 
-import io.airlift.units.Duration;
 import io.prestosql.client.StatementClient;
 
 import java.io.Closeable;
 import java.io.IOException;
-import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.atomic.AtomicBoolean;
 
-import static io.airlift.units.Duration.nanosSince;
 import static java.util.Collections.unmodifiableList;
 import static java.util.Objects.requireNonNull;
-import static java.util.concurrent.TimeUnit.SECONDS;
 
 public final class OutputHandler
         implements Closeable
 {
-    private static final Duration MAX_BUFFER_TIME = new Duration(3, SECONDS);
```

Comment:
`AlignedTablePrinter` prints a fixed width table, where it uses the maximum width of a cell in a batch to determine the width of the table. The goal of this code is to provide large batches of rows to the printer so that the width doesn't change every few rows. It also has an upper time limit for each batch in case the query is returning data slowly (which can happen for very selective queries).

Reply:
Keeping the numElements parameter in Guava's Queues.drain() API 100. So, we will either wait for 100 records or 3 seconds before draining the queue if the query is not finished.

Reply:
I think we should make it `1_000`. The number of lines on a screen is ~200 at a point of time. If we make it `10_000`, we will have to wait for 3 seconds before the user starts to see any output for a slow running query.

Reply:
Agreed, `1000` seems good.